front/app/containers/SearchPage/components/AggFilterInputUpdater.tsx

```diff
@@ -143,34 +134,82 @@ class AggFilterInputUpdater {
 
   getMinString(): string | undefined {
     // need to check for agg type once we start using this for more than date.
-    if (this.aggFilterInput.gte) {
+    if (this.input.gte) {
       return this.isDateAgg()
-        ? moment(this.aggFilterInput.gte)
+        ? moment(this.input.gte)
             .utc(false)
             .format('YYYY-MM-DD')
-        : this.aggFilterInput.gte;
+        : this.input.gte;
     }
   }
 
   getMaxString(): string | undefined {
     // need to check for agg type once we start using this for more than date.
-    if (this.aggFilterInput.lte) {
+    if (this.input.lte) {
       return this.isDateAgg()
-        ? moment(this.aggFilterInput.lte)
+        ? moment(this.input.lte)
             .utc(false)
             .format('YYYY-MM-DD')
-        : this.aggFilterInput.lte;
+        : this.input.lte;
     }
   }
 
   allowsMissing(): boolean {
-    return this.aggFilterInput!.includeMissingFields || false;
+    return this.input!.includeMissingFields || false;
   }
 
   toggleAllowMissing(): void {
-    this.aggFilterInput.includeMissingFields = !this.aggFilterInput
-      .includeMissingFields;
-    this.updateSearchParamsForAggFilterInput();
+    this.input.includeMissingFields = !this.input.includeMissingFields;
+    this.onUpdateFilter();
+  }
+}
+
+/**
+ * Responsible for updating aggs in the context of a search
+ */
+class AggFilterInputUpdater extends AggFilterInputUpdaterBase {
```

Comment:
most of how `AggFilterUpdater` works just got moved to the `Base` class. Probably could call it `AbstractAggFilterUpdater` since it can't be instantiated...